.pre-commit-config.yaml

```diff
@@ -0,0 +1,53 @@
+# pre-commit package installation is necessary to use pre-commit.
+# $ pip install pre-commit
+# $ pre-commit install
+
+default_language_version:
+  python: python3
+
+repos:
+  # Args are based on setup.cfg.
+  - repo: https://github.com/psf/black
+    rev: 24.8.0
+    hooks:
+      - id: black
+  - repo: https://github.com/pycqa/flake8
+    rev: 7.1.1
+    hooks:
+      - id: flake8
+        exclude: build|tutorial|docs/visualization_examples|docs/visualization_matplotlib_examples
+        args: [
+            "--max-line-length=99",
+            "--ignore=E203,E704,W503",
+            "--statistics",
+        ]
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.11.2
+    hooks:
+    - id: mypy
+      additional_dependencies: [
+        "mypy_boto3_s3",
+        "types-PyYAML",
+        "types-redis",
+        "types-setuptools",
+        "types-tqdm",
+        "typing_extensions>=3.10.0.0",
+      ]
```

Comment:
```suggestion
      additional_dependencies: [
        "alembic>=1.5.0",
        "colorlog",
        "numpy",
        "packaging>=20.0",
        "sqlalchemy>=1.3.0",
        "tqdm",
        "PyYAML",
        "mypy_boto3_s3",
        "types-PyYAML",
        "types-redis",
        "types-setuptools",
        "types-tqdm",
        "typing_extensions>=3.10.0.0",
      ]
```
I think we should also include the default dependencies of `optuna`, as mypy validates the types of all installed dependencies.